bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/handler/AVMFritzBaseBridgeHandler.java

```diff
@@ -222,208 +223,46 @@ public void addTemplateList(List<TemplateModel> templateList) {
     }
 
     /**
-     * Called from {@link FritzAhaUpdateCallback} to provide new values for things.
+     * Called from {@link FritzAhaUpdateCallback} to provide new devices.
      *
-     * @param deviceList list of device models
+     * @param deviceList list of devices
      */
-    public void addDeviceList(List<AVMFritzBaseModel> deviceList) {
-        for (Thing thing : getThing().getThings()) {
+    public void onDeviceListAdded(List<AVMFritzBaseModel> deviceList) {
+        getThing().getThings().stream().forEach(thing -> {
             AVMFritzBaseThingHandler handler = (AVMFritzBaseThingHandler) thing.getHandler();
             if (handler != null) {
                 Optional<AVMFritzBaseModel> optionalDevice = deviceList.stream()
                         .filter(it -> it.getIdentifier().equals(handler.getIdentifier())).findFirst();
```

Comment:
This logic could be optimized
```suggestion
        Map<String, AVMFritzBaseModel> deviceMap = deviceList.stream().collect(Collectors.toMap(it -> it.getIdentifier(), Functions.identity()));
        getThing().getThings().forEach(thing -> {
            AVMFritzBaseThingHandler handler = (AVMFritzBaseThingHandler) thing.getHandler();
            if (handler != null) {
                Optional<AVMFritzBaseModel> optionalDevice = Optional.ofNullable(deviceMap.get(handler.getIdentifier());
```